src/sdp.ts

```diff
@@ -1,109 +1,119 @@
+  const mhdecoded = decodeCerthash(certhash(ma))
+  const prefix = toSupportedHashFunction(mhdecoded.name)
+  const fp = mhdecoded.digest.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '')
+  const sdp = fp.match(/.{1,2}/g)
 
-  const fp = mhdecoded.digest.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
-  const fpSdp = fp.match(/.{1,2}/g)!.join(':');
+  if (sdp == null) {
+    throw invalidFingerprint(fp, ma.toString())
+  }
 
-  return [`${prefix.toUpperCase()} ${fpSdp.toUpperCase()}`, fp];
+  return [`${prefix.toUpperCase()} ${sdp.join(':').toUpperCase()}`, fp]
 }
 
-export function toSupportedHashFunction(name: multihashes.HashName): string {
+// Normalize the hash name from a given multihash has name
+export function toSupportedHashFunction (name: multihashes.HashName): string {
   switch (name) {
     case 'sha1':
       return 'sha-1'
     case 'sha2-256':
-      return 'sha-256';
+      return 'sha-256'
     case 'sha2-512':
-      return 'sha-512';
+      return 'sha-512'
     default:
-      throw unsupportedHashAlgorithm(name);
+      throw unsupportedHashAlgorithm(name)
   }
 }
 
-function ma2sdp(ma: Multiaddr, ufrag: string): string {
-  const IP = ip(ma);
-  const IPVERSION = ipv(ma);
-  const PORT = port(ma);
-  const [CERTFP, _] = certhashToFingerprint(ma);
+// Convert a multiaddr into a SDP
+function ma2sdp (ma: Multiaddr, ufrag: string): string {
+  const { host, port } = ma.toOptions()
+  const ipVersion = ipv(ma)
+  const [CERTFP] = ma2Fingerprint(ma)
+
   return `v=0
-o=- 0 0 IN ${IPVERSION} ${IP}
+o=- 0 0 IN ${ipVersion} ${host}
 s=-
-c=IN ${IPVERSION} ${IP}
+c=IN ${ipVersion} ${host}
 t=0 0
 a=ice-lite
-m=application ${PORT} UDP/DTLS/SCTP webrtc-datachannel
+m=application ${port} UDP/DTLS/SCTP webrtc-datachannel
 a=mid:0
 a=setup:passive
 a=ice-ufrag:${ufrag}
 a=ice-pwd:${ufrag}
 a=fingerprint:${CERTFP}
 a=sctp-port:5000
 a=max-message-size:100000
-a=candidate:1467250027 1 UDP 1467250027 ${IP} ${PORT} typ host\r\n`;
+a=candidate:1467250027 1 UDP 1467250027 ${host} ${port} typ host\r\n`
 }
 
-export function fromMultiAddr(ma: Multiaddr, ufrag: string): RTCSessionDescriptionInit {
+// Create an answer SDP from a multiaddr
+export function fromMultiAddr (ma: Multiaddr, ufrag: string): RTCSessionDescriptionInit {
   return {
     type: 'answer',
-    sdp: ma2sdp(ma, ufrag),
-  };
+    sdp: ma2sdp(ma, ufrag)
+  }
 }
 
-export function munge(desc: RTCSessionDescriptionInit, ufrag: string): RTCSessionDescriptionInit {
-  if (desc.sdp) {
-    desc.sdp = desc.sdp.replace(/\na=ice-ufrag:[^\n]*\n/, '\na=ice-ufrag:' + ufrag + '\n').replace(/\na=ice-pwd:[^\n]*\n/, '\na=ice-pwd:' + ufrag + '\n');
-    return desc;
-  } else {
-    throw invalidArgument("Can't munge a missing SDP");
+// Replace the ufrag and password values in a SDP
+export function munge (desc: RTCSessionDescriptionInit, ufrag: string): RTCSessionDescriptionInit {
+  if (desc.sdp === undefined) {
```

Comment:
```suggestion
  if (!desc.sdp) {
```

no?

Reply:
In general, the linter didn't like these type coercions and wanted explicit types.  When I turn this on, I get:
```
/js-libp2p-webrtc/src/sdp.ts:111:8: Unexpected nullable string value in conditional. Please handle the nullish/empty cases explicitly. [Error/@typescript-eslint/strict-boolean-expressions]
``` 

I'm going to resolve any other comments related to this.